bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,240 @@
+import org.openhab.binding.etherrain.internal.EtherRainException;
+import org.openhab.binding.etherrain.internal.api.EtherRainCommunication;
+import org.openhab.binding.etherrain.internal.api.EtherRainStatusResponse;
+import org.openhab.binding.etherrain.internal.config.EtherRainConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EtherRainHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Joe Inkenbrandt - Initial contribution
+ */
+public class EtherRainHandler extends BaseThingHandler {
+
+  private final Logger logger = LoggerFactory.getLogger(EtherRainHandler.class);
+
+  private EtherRainCommunication device = null;
+  private boolean connected = false;
+
+  @Nullable
+  private ScheduledFuture<?> updateJob;
+
+  /*
+   * Constructor class. Only call the parent constructor
+   */
+  public EtherRainHandler(Thing thing) {
+    super(thing);
+  }
+
+  @Override
+  public void handleCommand(ChannelUID channelUID, Command command) {
+    if (command.toString().equals("REFRESH")) {
+      updateBridge();
+      return;
+    }
+
+    else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_EXECUTE)) {
+      execute();
+      updateState(EtherRainBindingConstants.CHANNEL_ID_EXECUTE, OnOffType.OFF);
+    }
+
+    else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_CLEAR)) {
+      clear();
+      updateState(EtherRainBindingConstants.CHANNEL_ID_CLEAR, OnOffType.OFF);
+    } else {
+      logger.warn("Unhandled command: " + command.toFullString());
+    }
+
+  }
+
+  private boolean connectBridge() {
+    EtherRainConfiguration config = getConfigAs(EtherRainConfiguration.class);
+
+    logger.debug("Attempting to connect to Etherrain wtih config = (Host: {}, Port: {}, Password: {}, Refresh: {}).",
+        config.host, config.port, config.password, config.refresh);
+
+    device = new EtherRainCommunication(config.host, config.port, config.password);
+
+    EtherRainStatusResponse response;
+    try {
+      response = device.commandStatus();
+    } catch (EtherRainException | IOException e) {
+      updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+          "Could not create a connection to the EtherRain");
+      logger.debug("Could not open API connection to the EtherRain device. Exception received: {}", e.toString());
+      device = null;
+      updateStatus(ThingStatus.OFFLINE);
+      return false;
+    }
+    if (response == null) {
+      logger.debug("Command Status returned null");
+      device = null;
+      updateStatus(ThingStatus.OFFLINE);
+      return false;
+    }
+
+    updateStatus(ThingStatus.ONLINE);
+
+    return true;
+  }
+
+  private void startUpdateJob() {
+    logger.debug("Starting Etherrain Update Job");
+    updateJob = scheduler.scheduleWithFixedDelay(this::updateBridge, 0,
+        getConfigAs(EtherRainConfiguration.class).refresh, TimeUnit.SECONDS);
```

Comment:
If you store the config in a class field, you only have to call `getConfigAs` once.